### ISSUE TITLE
pnpm like installation mode to save space

### ISSUE THREAD

**Issue by jnoortheen** [python-poetry/poetry#3487]
- [x] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [x] I have searched the [documentation](https://python-poetry.org/docs/) and believe that my question is not covered.

## Feature Request

when I start a new project and start adding some development dependencies, the virtualenv quickly occupy ~100 MB of disk space, with datascience libraries, it becomes even GBs. 

can poetry have a mode where it installs a dependency only once and save disk space like pnpm ? 

I think it is easier than in npm space. Since Python development flow already uses virtualenv scripts to alter environment variables.

**Comment by finswimmer** [comment]
Hey,

this is currently out of scope for Poetry, because we use `virtualenv` under the hood to create a venv or use the one created by the user beforehand.

fin swimmer
